crates/agent/src/discovers/specs.rs

```diff
@@ -112,147 +114,141 @@ fn index_fetched_bindings<'a>(
+    resource_path_pointers: &[doc::Pointer],
+) -> anyhow::Result<(Vec<UsedBinding>, Changes, Changes)> {
+    assert!(
+        !resource_path_pointers.is_empty(),
+        "expected resource_path_pointers to be non-empty"
+    );
     let capture_prefix = capture_name.rsplit_once("/").unwrap().0;
 
-    let (fetched_bindings, interval, shards, auto_discover) = match fetched_capture {
-        Some(models::CaptureDef {
-            auto_discover,
-            endpoint: _,
-            bindings: fetched_bindings,
-            interval,
-            shards,
-            expect_pub_id: _, // The persisted model always has this set to None.
-            delete: _,        // The persisted model always has this set to false.
-        }) => (fetched_bindings, interval, shards, auto_discover),
-
-        None => (
-            Vec::new(),
-            models::CaptureDef::default_interval(),
-            models::ShardTemplate::default(),
-            Some(models::AutoDiscover {
-                add_new_bindings: true,
-                evolve_incompatible_collections: true,
-            }),
-        ),
-    };
-
-    let pointers = resource_path_pointers
-        .iter()
-        .map(|p| doc::Pointer::from_str(p.as_str()))
-        .collect::<Vec<_>>();
-
-    let fetched_bindings_by_path = if !pointers.is_empty() {
-        index_fetched_bindings(&pointers, &fetched_bindings)?
-    } else {
-        Default::default()
-    };
-
-    let mut capture_bindings = Vec::new();
-    let mut filtered_bindings = Vec::new();
+    let mut existing_bindings_by_path =
+        index_fetched_bindings(resource_path_pointers, &model.bindings)?;
+    let mut added_resources = BTreeMap::new();
+    let mut used_bindings = Vec::with_capacity(discovered_bindings.len());
 
+    let mut next_bindings = Vec::new();
     for discovered_binding in discovered_bindings {
         let Binding {
             recommended_name,
             resource_config_json,
+            document_schema_json,
+            key,
             ..
-        } = &discovered_binding;
-
-        let resource: serde_json::Value = serde_json::from_str(&resource_config_json).unwrap();
-
-        // Attempt to find a fetched binding such that the discovered resource
-        // spec is a strict subset of the fetched resource spec. In other
-        // words the fetched resource spec may have _extra_ locations,
-        // but all locations of the discovered resource spec must be equal.
-        let fetched_binding = if resource_path_pointers.is_empty() {
-            // TODO(phil): Legacy matching behavior, to be removed
-            fetched_bindings
-                .iter()
-                .filter(|fetched| {
-                    doc::diff(Some(&serde_json::json!(&fetched.resource)), Some(&resource))
-                        .is_empty()
-                })
-                .next()
+        } = discovered_binding;
+        // Don't use the deprecated `resource_path` on the `proto_flow::...::Binding` struct.
```

Comment:
I do think that we should consider the `resource_path` field to be deprecated. It's already been removed from the `Discovered` responses of all but 4 connectors (kafka, and 3 batch sql connectors). And `resource_path_pointers` are populated in `connector_tags` for every connector except for `materialize-slack`. The comment here is new, but we've considered this field to be deprecated for almost a year now. [This grafana logs query](https://estuary.grafana.net/explore?schemaVersion=1&panes=%7B%225fw%22:%7B%22datasource%22:%22grafanacloud-logs%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22count%20by%28image%29%20%28rate%28%7Bcluster%3D%5C%22env.estuary.combustable-cronut%5C%22,%20job%3D%5C%22flow%2Fagent%5C%22,%20container%3D%5C%22agent%5C%22%7D%20%7C%3D%20%60connector%20discovered%20response%20includes%20deprecated%20field%60%20%7C%20decolorize%20%7C%20regexp%20%60image_name%3D%28%3FP%3Cimage%3E%5B%5E%5C%5Cs%5D%2B%29%60%20%5B$__auto%5D%29%29%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22grafanacloud-logs%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-24h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1) shows the existing logs of connectors that still return `resource_path`.

I agree that there's still some unfinished business here, particularly for materialization connectors. But I think at least for capture connectors our plan to use `resource_path_pointers` instead of relying on returned `resource_path`s is still pretty reasonable, and is nearly fully realized.

See: #1291 and #1285 for some prior context

Reply:
Okay, I stand down here.